cmake/coverage.cmake

```diff
@@ -0,0 +1,20 @@
+check_cxx_compiler_flag("-0g" has_debug_optimization)
+if (has_debug_optimization)
+    # disable optimizations for better coverage stats
+    add_compile_options("-0g")
```

Comment:
I noticed you're building for release in the CI. Why not remove this and just run coverage on debug builds?

Reply:
Oh that's a good idea, done!